src/rapids_dependency_file_generator/rapids_dependency_file_generator.py

```diff
@@ -108,6 +110,31 @@ def make_dependency_file(
         )
     if file_type == str(OutputTypes.REQUIREMENTS):
         file_contents += "\n".join(dependencies) + "\n"
+    if file_type in (
+        str(OutputTypes.PYPROJECT_BUILD),
+        str(OutputTypes.PYPROJECT_DEPENDENCIES),
+        str(OutputTypes.PYPROJECT_OPTIONAL_DEPENDENCIES),
+    ):
+        # This file type needs to be modified in place instead of built from scratch.
+        with open(os.path.join(output_dir, name)) as f:
+            file_contents = tomlkit.load(f)
+
+        toml_deps = tomlkit.array()
+        for dep in dependencies:
+            toml_deps.add_line(dep)
+        toml_deps.add_line(indent="")
+        toml_deps.comment(
+            f"This list was generated by `{cli_name}`. To make changes, edit "
+            f"{relative_path_to_config_file} and run `{cli_name}`."
+        )
+        if file_type == str(OutputTypes.PYPROJECT_BUILD):
+            file_contents["build-system"]["requires"] = toml_deps
+        elif file_type == str(OutputTypes.PYPROJECT_DEPENDENCIES):
+            file_contents["project"]["dependencies"] = toml_deps
+        else:
+            # TODO: May want to make sure the optional-dependencies table exists
```

Comment:
Let’s address this TODO. Check for the key (or some other approach, like catching an exception).